interactive/dashboard/flow_diagram.py

```diff
@@ -119,153 +94,1008 @@ def name_included_reactions(included_reactions, reactions_json):
+                    for product in r['products']:
+                        species_nodes.update({product: {}})
+                        reaction_string = reaction_string + product + "_"
+                    if r['products'] == [] or len(r['products']) == 0:
+                        # reaction_string = reaction_string[:-2]
+                        print("* no products:", reaction_string)
+                    else:
+                        reaction_string = reaction_string[:-1]
+                    reactions_nodes.update({reaction_string: {}})
+                if species in r['products']:
+                    tmp_val = reactions_data['camp-data'][0]['reactions']
+                    contained_reactions.update(
+                        {tmp_val.index(r): {}})
+                    reaction_string = ""
+                    for reactant in r['reactants']:
+                        species_nodes.update({reactant: {}})
+                        reaction_string = reaction_string + reactant + "_"
+                    # remove last + and replace with →
+                    reaction_string = reaction_string[:-1] + "->"
+                    for product in r['products']:
+                        species_nodes.update({product: {}})
+                        reaction_string = reaction_string + product + "_"
+                    reaction_string = reaction_string[:-1]
+                    reactions_nodes.update({reaction_string: {}})
+    for speci in species_nodes:
+        if speci not in list_of_species:
+            species_colors.update({speci: "#94b8f8"})
+            species_sizes.update({speci: 20})
+    return (contained_reactions, species_nodes,
+            reactions_nodes, species_colors, species_sizes)
 
+
+# return list of raw widths for arrows. Also set new minAndMax
+def findReactionRates(reactions_nodes, df, start, end, csv_results_path_default=csv_results_path_default):
+    global minAndMaxOfSelectedTimeFrame
+    global userSelectedMinMax
+    global previous_vals
+    rates_cols = [x for x in df.columns if 'myrate' in x]
+    reactionsToAdd = []
+    for re in reactions_nodes:
+        # convert index of reaction into actual reaction name
+        reactionsToAdd.append(re)
+    rates = df[rates_cols]
+    first = 0
+    last = len(time_column_list(csv_results_path_default))-1
+    if start != 1:
+        first = time_column_list(csv_results_path_default).index(start)
+        last = time_column_list(csv_results_path_default).index(end)
+    first_and_last = rates.iloc[[first, last]]
+    difference = first_and_last.diff()
+    values = dict(difference.iloc[-1])
+    widths = {}
+    for key in values:
+        if key.split('.')[1].split("__")[1] in reactionsToAdd:
+            widths.update({key.split('.')[1].split("__")[1]: float(
+                str('{:0.3e}'.format(values[key])))})
+    return widths
+
+# get the actual min and max
+def minAndmax(reaction_nodes, quantities, widths):
+    min_val = 999999999999
+    max_val = 0
```

Comment:
Everywhere that we're using these, can we use `np.inf` and `-np.inf` instead?

Reply:
Or is that not possible because sometimes these get returned to the frontend?

Reply:
could we use [this](https://stackoverflow.com/a/42544543/5217293) instead?

Reply:
The problem with that is the max number in python is 9223372036854775807 and the max number in js is 9007199254740991 so there are concurrency issues i guess you would call it. I believe the front end relies on 999999999 and 0 as well, tho I could take a stab at changing it

Reply:
Then maybe a better way is to set these min and max values to the min and max of the data?